purchase_deposit_preserve_amount/models/account_move.py

```diff
@@ -0,0 +1,73 @@
+# Copyright 2025 Quartile (https://www.quartile.co)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    is_deposit = fields.Boolean(string="Is Deposit Move", readonly=True)
+
+    def _adjust_journal_item_balances_for_deposit(self):
+        for rec in self:
+            deposit_lines = rec.line_ids.filtered(
+                lambda line: line.display_type == "product"
+                and line.purchase_line_id.is_deposit
+            )
+            amount_diff = 0.0
+            for line in deposit_lines:
+                balance = line.purchase_line_id.invoice_lines.filtered(
+                    lambda l: l.move_id.state == "posted" and l.move_id.is_deposit
+                ).balance
+                amount_diff += balance + line.balance
+                line.with_context(skip_deposit_adjustment=True).balance = -1 * balance
```

Comment:
A minor point. I prefer early exits to reduce computation.
```suggestion
                line.with_context(skip_deposit_adjustment=True).balance = -1 * balance
            if not amount_diff:
                continue
```